src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs

```diff
@@ -346,7 +346,13 @@ private string CreateCompliantNameReusingPartialPrefixesAndSuffixes(string name)
             return FinishFixingName(name);
         }
 
-        private static string StripCommonPrefixes(string name, out string prefix)
+        public static string StripCommonPrefixes(string name)
+        {
+            StripCommonPrefixes(name, out _);
+            return name;
```

Comment:
❗️ The test _should_ have failed due to this mistake, but it did not. Prior to fixing this method, it would be good to update the test to a failing state so you can see that fixing this method fixes the test. (The changes to test and this method can be submitted as one commit; the separation is more to help you locally validate the change.)

Reply:
The tests were not failing because the desired changes were also being made in the `MakeCompliant` method called immediately after.  As a result, there are two lines of code before the `MakeCompliant` code that are no longer needed:
```csharp
 var trimmed = nameBasedOnArgument.TrimStart('_');
 BestNameForParameter = trimmed.Length > 0 ? trimmed.ToCamelCase() : nameBasedOnArgument;
```

I've updated this PR to remove those lines and added some tests to verify the behavior is still as expected.